### PR TITLE
Move basePath from OperationSpec.baseUrl to OperationSpec.path

### PR DESCRIPTION
**Pull request by daschult**
Resolves https://github.com/Azure/ms-rest-js/issues/113.

### CODE REVIEW

File: src/vanilla/Model/MethodTS.cs [hunk 1]
```diff
@@ -59,6 +59,45 @@ public override Parameter Add(Parameter item)
             return parameter;
         }
 
+        [JsonIgnore]
+        public string SchemeHostAndPort => (CodeModel as CodeModelTS).SchemeHostAndPort;
```
Comment (RikkiGibson) [inline src/vanilla/Model/MethodTS.cs]: If this expected to never fail I prefer a cast. Possibly a CodeModelTS property to contain that?

**Comment by RikkiGibson** [comment]
Is this going to satisfy both the use cases for URL/basePath? Or is this just part of that work?

**Comment by RikkiGibson** [comment]
Same CI error!

```sh
        Swagger LRO Happy BAT
          should work with Post200WithPayload:
      Uncaught AssertionError: expected '1' to be 1
```

Are you able to repro this locally @daschult ?

File: src/vanilla/Model/MethodTS.cs (hunk 1 above)
Comment (daschult) [reply src/vanilla/Model/MethodTS.cs]: Fair enough.

**Comment by daschult** [comment]
@RikkiGibson No, all of my tests pass fine. This is weird.

**Comment by daschult** [comment]
@RikkiGibson Yes, this should solve all of our basePath problems.

**Comment by daschult** [comment]
@RikkiGibson Just repro-ed the failed test. Update the `@microsoft.azure/autorest.testserver` dependency to  2.5.12. Fails like a charm. I'll look into this.

**Comment by RikkiGibson** [comment]
https://github.com/Azure/autorest.testserver/commit/4d11c867f46397a92a8497b7f8f7a7b726faff47

**Comment by RikkiGibson** [comment]
I would just change the `1` to `"1"`.

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
### Size decreased by 0.02% (-501 B)

**Pull request merged**